source/extensions/http/cache/file_system_http_cache/cache_eviction_thread.cc

```diff
@@ -0,0 +1,75 @@
+#include "source/extensions/http/cache/file_system_http_cache/cache_eviction_thread.h"
+
+#include "source/common/api/os_sys_calls_impl.h"
+#include "source/extensions/http/cache/file_system_http_cache/file_system_http_cache.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace FileSystemHttpCache {
+
+CacheEvictionThread::CacheEvictionThread(Thread::ThreadFactory& thread_factory)
+    : os_sys_calls_(Api::OsSysCallsSingleton::get()),
```

Comment:
as os_sys_calls_ is not used by this class directly, could we avoid holding onto it, and just pass that directly to the constructor of the FileSystemHttpCache?

Reply:
Aha, or not even pass it at all and just get it from the singleton since we're not using dependency injection of it in any tests.

And I definitely hope to remove os_sys_calls in favor of a file-specific abstraction anyway, once I'm done buffing that up. :)